src/malli/swagger.cljc

```diff
@@ -101,12 +101,12 @@
 (defmethod expand ::responses [_ v acc _]
   {:responses
    (into
-    (or (:responses acc) {})
-    (for [[status response] v]
-      [status (-> response
-                  (update :schema transform {:type :schema})
-                  (update :description (fnil identity ""))
-                  -remove-empty-keys)]))})
+     (or (:responses acc) {})
+     (for [[status response] (filter #(-> % second :schema) v)]
+       [status (-> response
+                   (update :schema transform {:type :schema})
```

Comment:
Instead of filtering out the responses without a schema, could you just skip the transform step for a nil schema? That way the other keys like `:description` would still end up in the swagger spec.